### ISSUE TITLE
How do I run non-gguf models like GPTQ ? 

### ISSUE THREAD

**Issue by tonypius** [KillianLucas/open-interpreter#322]
### Is your feature request related to a problem? Please describe.

I want to run the phind-codellama-v2-gptq model. How do i do that now ? I've already tried passing it into the --model parameter. But no luck, since its redirecting to the GGUF models.  

### Describe the solution you'd like

Ideally, we should be able to run all models, not just GGUF. More importantly, we should be able to utilise already running models on TGI or vllm api endpoints.

**Comment by PaulConyngham** [comment]
using this pull request:

https://github.com/KillianLucas/open-interpreter/pull/311

**Comment by abhinavkulkarni** [comment]
+1 for adding support for HuggingFace TGI. TGI is probably the most popular framework for LLM serving and it would be good if I could specify a remote TGI server endpoint and run local interpreter locally on my laptop.

**Comment by Veda-Swaroop** [comment]
**Problem Description:**

I would like to suggest the same feature request here. I tried open interpreter. It is an amazing project. But GGUF models don't run on GPU without problems. I believe GGUF models are meant to be run on CPU and not on GPU. Whereas GPTQ models are designed to run on GPU. On a limited hardware, running even a 7B Llama2 GGUF model is very difficult as it consumes most of CPU power. I request Devs. to look into this issue and enable GPTQ model support for this project. 

**Alternatives I tried:**
I tried Oobabooga's text generation web Ui, which is another great project. It supports GPTQ models. I was able to run models using that. But it is like chatting with a chatbot and not completely functional compared to open interpreter.
